File: refsol/task-scheduler.js
export function leastInterval(tasks, n) {
  const counts = {};
  for (const t of tasks) counts[t] = (counts[t] || 0) + 1;
  const values = Object.values(counts);
  const maxCount = Math.max(...values);
  const numMax = values.filter((c) => c === maxCount).length;
  const slots = (maxCount - 1) * (n + 1) + numMax;
  return Math.max(slots, tasks.length);
}

File: refsol/task-scheduler.py
from collections import Counter


def leastInterval(tasks, n):
    counts = Counter(tasks)
    max_count = max(counts.values())
    num_max = sum(1 for c in counts.values() if c == max_count)
    slots = (max_count - 1) * (n + 1) + num_max
    return max(slots, len(tasks))
